frameworks/hdfs/src/main/dist/svc.yml

```diff
@@ -361,9 +381,17 @@ pods:
         env:
           DATANODE: true
         # If HDFS does not report a data node to be live, we don't consider the data node to be ready yet
-        # TODO: LIBPROCESS_IP gets overriden to 0.0.0.0 in overlay networking so this readiness-check needs to be fixed
         readiness-check:
+          {{#KERBEROS_ENABLED}}
+          cmd: >
+            export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jre*/) &&
+            KRB5_CONFIG={{JRE_VERSION}}/lib/security/krb5.conf kinit -k -t ${MESOS_SANDBOX}/hdfs.keytab ${KERBEROS_PRIMARY}/data-${POD_INSTANCE_INDEX}-node.${FRAMEWORK_NAME}.autoip.dcos.thisdcos.directory@${KERBEROS_REALM} &&
```

Comment:
Use `{{FRAMEWORK_HOST}}` and `{{TASK_NAME}}` for the FQDN.

Reply:
Good catch